### PR TITLE
General Server Cleanup

### LINKED ISSUE #572: "unable to publish ballots to audit" failure
```16:41:46,523  INFO corla:91 - county 3 initially estimated to audit 294 ballots
16:41:46,599  INFO corla:242 - ASM event COUNTY_START_AUDIT_EVENT caused transition to COUNTY_AUDIT_UNDERWAY
16:41:46,603  INFO corla:242 - ASM event AUDIT_BOARD_START_AUDIT_EVENT caused transition to AUDIT_BOARD_SIGNED_OUT
java.lang.IllegalArgumentException: invalid sequence bounds
	at us.freeandfair.corla.controller.ComparisonAuditController.computeBallotOrder(ComparisonAuditController.java:171)
	at us.freeandfair.corla.controller.ComparisonAuditController.initializeAuditData(ComparisonAuditController.java:248)
	at us.freeandfair.corla.endpoint.PublishBallotsToAudit.endpoint(PublishBallotsToAudit.java:90)
	at us.freeandfair.corla.Main.lambda$activateEndpoints$5(Main.java:253)
	at spark.RouteImpl$1.handle(RouteImpl.java:72)
	at spark.http.matching.Routes.execute(Routes.java:61)
	at spark.http.matching.MatcherFilter.doFilter(MatcherFilter.java:130)
	at spark.embeddedserver.jetty.JettyHandler.doHandle(JettyHandler.java:50)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1568)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:564)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:317)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:110)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:128)
	at org.eclipse.jetty.util.thread.Invocable$InvocableExecutor.invoke(Invocable.java:222)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:294)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:199)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:672)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:590)
	at java.lang.Thread.run(Thread.java:745)
16:41:46,620 ERROR corla:443 - error 500 server error on /ballots-to-audit/publish: could not set ballot list for county 4 by stateadmin1 from localhost:8888
```

### PR DESCRIPTION
**Pull request by dmzimmerman**
Among other things, excises the "loop 3 times" equation from the system, and addresses some minor issues. 

Closes #572

### CODE REVIEW

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/StartAuditRound.java [hunk 1]
```diff
@@ -140,7 +155,7 @@ public String startRoundOne(final Request the_request, final Response the_respon
             asm.stepEvent(COUNTY_START_AUDIT_EVENT);
             final ASMEvent audit_event;
             if (asm.currentState().equals(CountyDashboardState.COUNTY_AUDIT_UNDERWAY) &&
-                cdb.comparisonAudits().isEmpty()) {
+                (cdb.comparisonAudits().isEmpty() || cdb.estimatedBallotsToAudit() == 0)) {
```
Comment (ranweiler) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/StartAuditRound.java]: In other places, we are checking that the ballots to audit is `> 0`. Could the result of `cdb.estimatedBallotsToAudit()` ever be negative, or was that fixed when you removed the the three-loop algorithm?

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/PublishBallotsToAudit.java [hunk 2]
```diff
@@ -1,132 +0,0 @@
-/*
```
Comment (ranweiler) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/PublishBallotsToAudit.java]: We're still using this endpoint in `master`, so we should change that before we merge this commit.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/StartAuditRound.java (hunk 1 above)
Comment (dmzimmerman) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/StartAuditRound.java]: I will check on that when I check on the issue Neal is still having.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/PublishBallotsToAudit.java (hunk 2 above)
Comment (dmzimmerman) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/PublishBallotsToAudit.java]: Yup

Comment (ranweiler) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/PublishBallotsToAudit.java]: Okay, as of the latest `sprint` branch, we no longer use this endpoint. We're safe to merge after we merge #559.

Comment (ranweiler) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/PublishBallotsToAudit.java]: #559 has landed, so this deletion will no longer break anything.

**Pull request merged**